src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -22,11 +22,11 @@ import { SavedObjectsErrorHelpers } from '../saved_objects';
 import { SavedObjectsClientContract } from '../saved_objects/types';
 import { Logger } from '../logging';
 import { createOrUpgradeSavedConfig } from './create_or_upgrade_saved_config';
-import { IUiSettingsClient, UiSettingsParams } from './types';
+import { IUiSettingsClient, UiSettingsParams, UiSettingsType } from './types';
 import { CannotOverrideError } from './ui_settings_errors';
 
 export interface UiSettingsServiceOptions {
-  type: string;
+  type: UiSettingsType;
```

Comment:
Is this really the same `UiSettingsType` here as in `UiSettingsParams` though ? The only value used when creating the client (such as in `src/core/server/ui_settings/ui_settings_service.ts`) seems to be `config` which is not a part of the enum and forced to add an hard-cast in both the service and the tests. Shouldn't either this stay a `string` or add `config` to the `UiSettingsType` enum?

I'm not an expert on this part of the code, maybe @restrry you could help me here?

Reply:
I'm seeing the same thing and reverting the change (which is clearly wrong). Perhaps this could use a more specific type but that doesn't have to happen in this PR.

Reply:
good for me

Reply:
ok with reverted to string 👍